crates/wasmprinter/src/operator.rs

```diff
@@ -238,6 +238,11 @@ impl<'a, 'b> PrintOperator<'a, 'b> {
         self.printer.print_idx(&self.state.core.type_names, idx)
     }
 
+    fn field_index(&mut self, idx: u32) -> Result<()> {
+        write!(&mut self.printer.result, "{idx}")?;
+        Ok(())
+    }
```

Comment:
Would you be up for threading field names through here? (or is there not a subsection of the `name` custom section to specify field names yet?)

Reply:
There is a fields subsection of the name custom section in the spec appendix, but we don't parse it at all in `wasmprinter` yet. I'd prefer to leave this for a follow up PR.